neqo-transport/src/connection/mod.rs

```diff
@@ -2003,6 +1996,10 @@ impl Connection {
 
         #[cfg(test)]
         {
+            // Need to check if the previous `write_frames` call filled the packet.
+            if builder.is_full() {
+                return;
+            }
```

Comment:
Maybe we don't need to for tests.  The test code can check for overflow... and even overflow if it wants.